source/v5/mqtt5_to_mqtt3_adapter.c

```diff
@@ -2184,10 +2184,14 @@ static void s_aws_mqtt5_to_mqtt3_adapter_subscribe_completion_fn(
     const struct aws_mqtt5_packet_suback_view *suback,
     int error_code,
     void *complete_ctx) {
-    (void)suback;
 
     struct aws_mqtt5_to_mqtt3_adapter_operation_subscribe *subscribe_op = complete_ctx;
     struct aws_mqtt_client_connection_5_impl *adapter = subscribe_op->base.adapter;
+    size_t reason_code_count = 0;
+
+    if (suback != NULL) {
+        reason_code_count = suback->reason_code_count;
+    }
```

Comment:
Probably move this down to the `granted_qos` part, so we only need to do the check once. 
```
if (suback == NULL) {
            granted_qos = AWS_MQTT_QOS_FAILURE;
        }
```

Reply:
Added it twice so i don't change the scope of the variable granted_qos